### PR TITLE
Refactored for SuMo update

### PR DESCRIPTION
**Pull request by Drowze**
Simple refactor for easier SuMo update.

I tried to include SuMo myself but, I am not sure why, the bottom screen is going red and the 3DS freezes whenever I try to open my Pokémon Moon game.
If I can have some help on that, I may try to do the update myself on this PR.

### CODE REVIEW

File: source/main.c [hunk 1]
```diff
@@ -183,11 +183,25 @@ Result loadCode()
...
+            ser_addr = search_string("https://3ds1-fushigi.pokemon-gl.com/api/", 0x490000, 0x100000, paramblk->code_data);
+            boss1_addr = search_string("https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s", 0x490000, 0x100000, paramblk->code_data);
+            boss2_addr = search_string("https://npfl.c.app.nintendowifi.net/p01/filelist/", 0x490000, 0x100000, paramblk->code_data);
+        }
+        else if(tid_short == 0x0011C400 || 0x0011C500) { // OR || AS
+            ser_addr = search_string("https://3ds2-fushigi.pokemon-gl.com/api/", 0x4E0000, 0x100000, paramblk->code_data);
+            boss1_addr = search_string("https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s", 0x4E0000, 0x100000, paramblk->code_data);
+            boss2_addr = search_string("https://npfl.c.app.nintendowifi.net/p01/filelist/", 0x4E0000, 0x100000, paramblk->code_data);
+        }
+        // else { // (tid_short == 0x00164800 || 0x00175E00) aka SUN || MOON
```
Comment (srwgin) [inline source/main.c:201]: +        else if(tid_short == 0x00164800 || 0x00175E00) { // SUN || MOON
 +            ser_addr = search_string("https://3ds3-fushigi.pokemon-gl.com/api/", 0x510000, 0x100000, paramblk->code_data);
 +            boss1_addr = search_string("https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s", 0x510000, 0x100000, paramblk->code_data);
 +            boss2_addr = search_string("https://npfl.c.app.nintendowifi.net/p01/filelist/", 0x510000, 0x100000, paramblk->code_data);

Comment (srwgin) [reply source/main.c:201]: but there is another https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s/%s
I do not know how do deal with it
maybe boss1_addr will replace 2 place

Comment (srwgin) [reply source/main.c:201]: if you want to use MysteryMachine-Patcher in hbl with sun moon
you also should use this
https://github.com/ihaveamac/9.6-dbgen-xmls

**Comment by shinyquagsire23** [comment]
Looks good, the reason Sun and Moon don't work however is because 11.2 added PASLR for them, and they have seed encryption so I think you need those XMLs mentioned by @srwgin as well. Otherwise I'll take it I guess, might extend it myself for when it _does_ work with HANS.

EDIT: Just noticed @srwgin had some code to make it work, I have an idea of how to get it working with the identical search phrase though.

**Pull request merged**